### ISSUE TITLE
{x = y} should be allowed sometimes

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#970]
quick-lint-js rejects the following legal code with "use ':' instead of '=' in object literals" (E0253):
```javascript
let x;
({x = null} = {});
```

**Comment by strager** [comment]
Fixed in version [2.16.0](https://quick-lint-js.com/releases/#2.16.0).
